### PR TITLE
[FanControl cluster] Add missing definitions for Fan Control cluster

### LINKED ISSUE #17409: Implement the missing parts of FanControl cluster
#### Problem
(https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/FanControl.adoc)

 #### Proposed Solution
<suggested fix, suggested enhancement>

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* This is reported from vendor integration and some fields in FanControl cluster are missing
* 
* Fixes #17409

#### Change overview
Add missing definitions for Fan Control cluster

#### Testing
How was this tested? (at least one bullet point required)
```
yufengwang@yufengwang:~/connectedhomeip/out/debug/standalone$ ./chip-tool fancontrol write percent-setting 60 12344321 1
....
[1649983987.182935][2127344:2127349] CHIP:DMG: WriteResponseMessage =
[1649983987.182950][2127344:2127349] CHIP:DMG: {
[1649983987.182962][2127344:2127349] CHIP:DMG: 	AttributeStatusIBs =
[1649983987.182980][2127344:2127349] CHIP:DMG: 	[
[1649983987.182994][2127344:2127349] CHIP:DMG: 		AttributeStatusIB =
[1649983987.183007][2127344:2127349] CHIP:DMG: 		{
[1649983987.183019][2127344:2127349] CHIP:DMG: 			AttributePathIB =
[1649983987.183034][2127344:2127349] CHIP:DMG: 			{
[1649983987.183048][2127344:2127349] CHIP:DMG: 				Endpoint = 0x1,
[1649983987.183062][2127344:2127349] CHIP:DMG: 				Cluster = 0x202,
[1649983987.183084][2127344:2127349] CHIP:DMG: 				Attribute = 0x0000_0002,
[1649983987.183098][2127344:2127349] CHIP:DMG: 			}
[1649983987.183116][2127344:2127349] CHIP:DMG: 				
[1649983987.183131][2127344:2127349] CHIP:DMG: 			StatusIB =
[1649983987.183145][2127344:2127349] CHIP:DMG: 			{
[1649983987.183161][2127344:2127349] CHIP:DMG: 				status = 0x00 (SUCCESS),
[1649983987.183178][2127344:2127349] CHIP:DMG: 			},
[1649983987.183193][2127344:2127349] CHIP:DMG: 			
[1649983987.183205][2127344:2127349] CHIP:DMG: 		},
[1649983987.183222][2127344:2127349] CHIP:DMG: 		
[1649983987.183234][2127344:2127349] CHIP:DMG: 	],
[1649983987.183252][2127344:2127349] CHIP:DMG: 	
[1649983987.183266][2127344:2127349] CHIP:DMG: 	InteractionModelRevision = 1
[1649983987.183277][2127344:2127349] CHIP:DMG: }

yufengwang@yufengwang:~/connectedhomeip/out/debug/standalone$ ./chip-tool fancontrol read percent-setting 12344321 1
....
[1649984000.163798][2127363:2127368] CHIP:DMG: ReportDataMessage =
[1649984000.163813][2127363:2127368] CHIP:DMG: {
[1649984000.163825][2127363:2127368] CHIP:DMG: 	AttributeReportIBs =
[1649984000.163843][2127363:2127368] CHIP:DMG: 	[
[1649984000.163855][2127363:2127368] CHIP:DMG: 		AttributeReportIB =
[1649984000.163873][2127363:2127368] CHIP:DMG: 		{
[1649984000.163886][2127363:2127368] CHIP:DMG: 			AttributeDataIB =
[1649984000.163899][2127363:2127368] CHIP:DMG: 			{
[1649984000.163913][2127363:2127368] CHIP:DMG: 				DataVersion = 0x3e7f1062,
[1649984000.163928][2127363:2127368] CHIP:DMG: 				AttributePathIB =
[1649984000.163942][2127363:2127368] CHIP:DMG: 				{
[1649984000.163956][2127363:2127368] CHIP:DMG: 					Endpoint = 0x1,
[1649984000.163971][2127363:2127368] CHIP:DMG: 					Cluster = 0x202,
[1649984000.163985][2127363:2127368] CHIP:DMG: 					Attribute = 0x0000_0002,
[1649984000.164004][2127363:2127368] CHIP:DMG: 				}
[1649984000.164019][2127363:2127368] CHIP:DMG: 					
[1649984000.164034][2127363:2127368] CHIP:DMG: 					Data = 60, 
[1649984000.164048][2127363:2127368] CHIP:DMG: 			},
[1649984000.164063][2127363:2127368] CHIP:DMG: 			
[1649984000.164076][2127363:2127368] CHIP:DMG: 		},
[1649984000.164093][2127363:2127368] CHIP:DMG: 		
[1649984000.164105][2127363:2127368] CHIP:DMG: 	],
[1649984000.164122][2127363:2127368] CHIP:DMG: 	
[1649984000.164136][2127363:2127368] CHIP:DMG: 	SuppressResponse = true, 
[1649984000.164148][2127363:2127368] CHIP:DMG: 	InteractionModelRevision = 1
[1649984000.164160][2127363:2127368] CHIP:DMG: }
[1649984000.164264][2127363:2127368] CHIP:TOO: Endpoint: 1 Cluster: 0x0000_0202 Attribute 0x0000_0002 DataVersion: 1048514658
[1649984000.164319][2127363:2127368] CHIP:TOO:   percent setting: 60
```

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/silabs/ha.xml [hunk 1]
```diff
@@ -53,11 +53,61 @@ limitations under the License.
     <define>FAN_CONTROL_CLUSTER</define>
     <client tick="false" init="false">true</client>
     <server tick="false" init="false">true</server>
-    <attribute side="server" code="0x0000" define="FAN_CONTROL_FAN_MODE" type="ENUM8" min="0x00" max="0x06" writable="true" default="0x05" optional="false">fan mode</attribute>
-    <!-- FAN_MODE -->
-    <attribute side="server" code="0x0001" define="FAN_CONTROL_FAN_MODE_SEQUENCE" type="ENUM8" min="0x00" max="0x04" writable="true" default="0x02" optional="false">fan mode sequence</attribute>
-    <!-- FAN_MODE_SEQUENCE -->
+    <attribute side="server" code="0x0000" define="FAN_MODE" type="ENUM8" min="0x00" max="0x06" writable="true" default="0x00" optional="false">fan mode</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/silabs/ha.xml:56]: Why is the type here not FanModeType or FanModeEnum (whatever the enum should actually be called)?  Similar for FanModeSequence....

File: src/app/zap-templates/zcl/data-model/silabs/ha.xml [hunk 2]
```diff
@@ -53,11 +53,61 @@ limitations under the License.
...
+    <attribute side="server" code="0x0001" define="FAN_MODE_SEQUENCE" type="ENUM8" min="0x00" max="0x04" writable="true" default="0x02" optional="false">fan mode sequence</attribute>
+    <attribute side="server" code="0x0002" define="PERCENT_SETTING" type="INT8U" min="0x00" max="0x64" writable="true" default="0x00" optional="false">percent setting</attribute>
+    <attribute side="server" code="0x0003" define="PERCENT_CURRENT" type="INT8U" min="0x00" max="0x64" writable="false" default="0x00" optional="false">percent current</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/silabs/ha.xml:59]: Might have been more readable with `max="100"`, as the spec has it...

**Review by bzbarsky-apple — COMMENTED** [review]
This does not seem to implement any of the things the spec says have to happen when the writable attributes are set.  Is there a followup tracking that?

**Comment by yufengwangca** [comment]
> implement any of the things the spec says have to happen when the writable attributes are set

https://github.com/project-chip/connectedhomeip/issues/17577

**Pull request merged**